distributed/worker_state_machine.py

```diff
@@ -1505,10 +1515,10 @@ def _ensure_communicating(self, *, stimulus_id: str) -> RecsInstrs:
             )
 
             self.in_flight_workers[worker] = to_gather_keys
-            self.comm_nbytes += total_nbytes
+            self.comm_incoming_bytes += total_nbytes
             if (
-                len(self.in_flight_workers) >= self.total_out_connections
-                and self.comm_nbytes >= self.comm_threshold_bytes
+                self.comm_incoming_count >= self.total_out_connections
```

Comment:
After implementing those changes, I am not a huge fan of `comm_incoming_count`. This line illustrates why: 
Why do we compare `comm_incoming_count` to `total_out_connections`? How are they related? 

`comm_incoming_count` still feels vaguely defined to me. This might be because `comm_` feels like a namespacing prefix for everything communications-related, so what are we counting exactly in that namespace that's incoming? I might just need to get more used to the fact that when we're talking about `comm_`, we tend to mean ` Comm` object. Then again, that `Comm` object feels equivalent to a connection, so why is the comm incoming while the  connection is outgoing? Should we get `transfer` back into the naming here?

Reply:
Happy to rename `comm` to `transfer` everywhere.
`comm_incoming_count` -> `transfer_incoming_count_current`
`total_out_connections` -> `transfer_incoming_count_max`

Reply:
I've in the meantime come up with a less invasive strategy that lets us keep the comm naming as is, I think basically all I want is to rename `total_{in|out}_connections` to `comm_{incoming|outgoing}_limit`. No need to shift over to transfer completely, just evade the term _connection_ when talking about comms.

Reply:
Done